core/controllers/ingest.py

```diff
@@ -69,6 +75,40 @@ def ingest(body: dict, excel_file: FileStorage) -> Response:
     return jsonify(success_payload)
 
 
+def process_validation_failures(validation_failures):
+    internal_failures = [failure for failure in validation_failures if isinstance(failure, InternalValidationFailure)]
+    user_failures = [failure for failure in validation_failures if isinstance(failure, UserValidationFailure)]
+
+    if internal_failures:
+        failure_uuid = save_failed_submission(g.excel_file)
+        current_app.logger.error(
+            f"Internal ingest exception - failure_id={failure_uuid} internal_failures: {internal_failures}"
+        )
+        return {
+            "detail": "Internal ingest exception.",
+            "id": failure_uuid,
+            "status": 500,
+            "title": "Internal Server Error",
+            "internal_errors": internal_failures,
+        }, 500
+
+    # create messaging object here and handle messages
+    messager = _messaging_class_factory("Towns Fund")  # TODO: Replace with fund type from ingest params
```

Comment:
I like the use of the factory pattern here. 

Although, this shouldn't be private if we're importing it from the `message` module